### ISSUE TITLE
Restructure ext/ folder

### ISSUE THREAD

**Issue by salkinium** [roboterclubaachen/xpcc#56]
Here is what I would like to see:

```
ext
├── README.md
├── SConscript
├── cmsis
│   ├── Include/*.h
│   └── DSP_Lib/Source/*.c
├── st
│   ├── STM32F0xx/Include/*.h
│   ├── STM32F1xx/Include/*.h
│   ├── STM32F2xx/Include/*.h
│   ├── STM32F3xx/Include/*.h
│   ├── STM32F4xx/Include/*.h
│   └── STM32F7xx/Include/*.h
├── nxp
│   └── lpc11xx
│       ├── Include/*.h
│       └── driver/*.{h,c}
├── fatfs/*
└── freertos/*
```

Explicitly the `lpc13xx`, `lpc17xx` and `stm32f10x_lib` folders are removed, since we don't support or use them anyway.

For the `lpc11xx` we use the ROM drivers for our CAN driver implementation, that's why we need the `driver/` folder.

Should we remove the `Include/` folders in all of the `STM32F*xx` folders? And rename them all `stm32f*xx`?

cc @ekiwi @dergraaf @strongly-typed

**Comment by salkinium** [comment]
Ok, some more thoughts after implementing this in #63.

Technically, we can include everything in `ext/` manually as well, not just as a global include.

``` cpp
#include <cmsis/Include/core_cm0.h>
#include <nxp/lpc11xx/driver/rom_drivers.h>
```

Maybe we should follow the capitalization convention of the source?

```
ext
├── README.md
├── SConscript
├── CMSIS
│   ├── Include/*.h
│   └── DSP_Lib/Source/*.c
├── STM32F0xx/Include/*.h
├── STM32F1xx/Include/*.h
├── STM32F2xx/Include/*.h
├── STM32F3xx/Include/*.h
├── STM32F4xx/Include/*.h
├── STM32F7xx/Include/*.h
├── LPC11xx
│       ├── Include/*.h
│       └── driver/*.{h,c}
├── fatfs/*
└── freertos/*
```

This would result in:

``` cpp
#include <CMSIS/Include/core_cm0.h>
#include <LPC11xx/driver/rom_drivers.h>
```

Maybe I don't really like this, and the way it was originally proposed is better structured.

**Comment by salkinium** [comment]
Closed via #63.
